Source/ASTextNode2.mm

```diff
@@ -323,10 +320,17 @@ - (NSArray *)exclusionPaths
   return _textContainer.exclusionPaths;
 }
 
-- (void)prepareAttributedString:(NSMutableAttributedString *)attributedString
+- (NSAttributedString *)l_preparedAttributedText
 {
-  ASLockScopeSelf();
+  if (_preparedAttributedText) {
+    return _preparedAttributedText;
+  }
+  
+  if (!_attributedText) {
```

Comment:
Move this condition above the other one, and instead of early return, just do _preparedAttributedText = [[NSAttributedString alloc] init];

Then it will still early return, and _preparedAttributedText will be set.